storage/range.go

```diff
@@ -387,11 +387,16 @@ func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp) error
 	return nil
 }
 
-// verifyLeaderLease checks whether the requesting replica (by raft
-// node ID) holds the leader lease covering the specified timestamp.
-func (r *Range) verifyLeaderLease(originRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) bool {
-	l := r.getLease()
-	return originRaftNodeID == l.RaftNodeID && timestamp.Less(l.Expiration)
+// verifyLeaderLease checks whether the requesting replica (by raft node ID)
+// holds the leader lease covering the specified timestamp.
+func verifyLeaderLease(lease *proto.Lease, replicaRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) (bool, bool) {
```

Comment:
the check in `updateStoreStatus` looks ok. The thinking is that if someone holds the leader lease for the range, then that range (as a Raft group) is doing well.